docs/source/tutorials/playing-command.rst

```diff
@@ -0,0 +1,150 @@
+=========================
+Playing with your command
+=========================
+
+Now that you have started :doc:`your first plugin <first-plugin>`, maybe you
+want to write a more interesting command than the basic ``.hello`` one. Not
+that there is anything wrong with that command! The emoticons plugin is
+composed of commands like this one::
+
+    @plugin.command('shrug')
+    @plugin.action_command('shrugs')
+    def shrug(bot, trigger):
+        bot.say('¯\\_(ツ)_/¯')
+
+Which is one of the maintainers' favorite commands to use. However, let's see
+if we can do something a bit more *complex* than that.
+
+
+Greeting a user by their name
+=============================
+
+Have you noticed that a :ref:`plugin callable <Plugin callables>` takes **two
+arguments?** The first one is the ``bot``, i.e. an instance of Sopel (wrapped
+for your convenience) that you can use to
+:doc:`interact with the bot </plugin/bot>`.
+
+In the previous tutorial, we used ``bot.reply``, which is convenient but not
+always what you want. Maybe you want the bot to say this::
+
+    <YourNick> .hello
+    <Sopel> Hello YourNick, have a nice day!
+
+For that, you need **the second argument**: the ``trigger``. It is an object
+with all the information you need about the message that triggered the
+callable, such as the **message** itself, the **channel**, the type of message,
+etc. and what we need for now is the
+:attr:`trigger.nick <sopel.trigger.Trigger.nick>` attribute::
+
+    from sopel import plugin
+
+    @plugin.command('hello')
+    def hello(bot, trigger):
+        """Say Hello <user>, have a nice day!"""
+        bot.say(f'Hello {trigger.nick}, have a nice day!')
+
+.. important::
+
+    If you want to test this with your bot, and your bot is already running,
+    you need to restart the bot so it will load the new version of your plugin.
+
+.. seealso::
+
+    You can learn much more about the :class:`trigger <sopel.trigger.Trigger>`
+    object by reading its documentation.
+
+
+Command with arguments
+======================
+
+The trigger object can do much more for you: if a user adds arguments to the
+command, like ``.hello morning``, you can detect and use that argument::
+
+    from sopel import plugin
+
+    @plugin.command('hello')
+    def hello(bot, trigger):
+        """Say Hello <user>, have a nice day!"""
+        # groups 3 to 6 are the first, second, third, and fourth command arg
+        when = trigger.group(3)
```

Comment:
I think some readers will look at this and think _"huh, what about groups 1 and 2?!"_ so it's probably worth mentioning them even though this example doesn't use them.

An example that would use group 2 could be something like:

```
<YourNick> .shout Hello, is this thing on?
<Sopel> HELLO, IS THIS THING ON?
```

but I think that additional comments will suffice for the careful reader.

```suggestion
        # group 1 is the name of the command that was triggered
        # group 2 is the entire rest of the message
        # groups 3 to 6 are the first, second, third, and fourth command arg
        when = trigger.group(3)
```

Reply:
I went back and forth on that one. Ultimately, it is way too confusing without the mention of what group 1 and 2 are doing.

Reply:
(and no, I'm not happy about it... but that's better that way)